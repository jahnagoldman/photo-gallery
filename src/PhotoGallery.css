.NoPhotos {
  display: flex;
  font-family: Helvetica;
  font-size: 40px;
  font-weight: bold;
  justify-content: center;
  margin-top: 40px;
}