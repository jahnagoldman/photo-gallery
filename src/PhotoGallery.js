import React, { useState } from 'react';
import { useSwipeable } from 'react-swipeable'
import { PropTypes } from 'prop-types';
import Photo from './Photo';
import './PhotoGallery.css';

function PhotoGallery(props) {
  const [index, setIndex] = useState(0);

  const handlers = useSwipeable({
    onSwipedLeft: () => goToNext(),
    onSwipedRight: () => goToPrevious(),
    preventDefaultTouchmoveEvent: true,
    trackMouse: true
  });

  function goToNext () {
    isLastImage() ? setIndex(0) : setIndex(index + 1);
  }

  function goToPrevious () {
    isFirstImage() ? setIndex(props.images.length - 1) : setIndex(index - 1);
  }

  function isLastImage () {
    return index === props.images.length - 1;
  }

  function isFirstImage () {
    return index === 0;
  }

  return (
    <div {...handlers}>
      {!props.images || props.images.length === 0
        ? <div className="NoPhotos">No photos!</div>
        : <Photo handlePhotoClick={() => goToNext()} image={props.images[index]}/>
      }
    </div>
  );
}

PhotoGallery.propTypes = {
  images: PropTypes.array
};

export default PhotoGallery;
