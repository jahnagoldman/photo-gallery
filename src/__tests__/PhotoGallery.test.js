import React from 'react';
import PhotoGallery from '../PhotoGallery';
import { act } from 'react-dom/test-utils';
import { render, unmountComponentAtNode } from "react-dom";

describe('PhotoGallery', () => {
  let container = null;

  beforeEach(() => {
    container = document.createElement("div");
    document.body.appendChild(container);
  });
      
  afterEach(() => {
    unmountComponentAtNode(container);
    container.remove();
    container = null;
  });
      
  it('renders without images', () => {
    act(() => {
        render(<PhotoGallery />, container);
    });

    expect(container).toMatchSnapshot()
    expect(container.textContent).toBe('No photos!');
  });
      
  it('renders with empty list of images', () => {
    act(() => {
        render(<PhotoGallery images={[]} />, container);
    });

    expect(container).toMatchSnapshot();
    expect(container.textContent).toBe('No photos!');
  })
    
  it('renders with images', () => {
    act(() => {
      render(<PhotoGallery images={[{'url': 'abc.com', 'caption': 'hello'}, {'url': 'hey.com', 'caption': 'cool'}]} />, container);
    });
    
    expect(container).toMatchSnapshot();
    expect(container.textContent).toBe('hello');
    expect(container.querySelector('img')).toBeTruthy();
  })
      
  it('clicking the image takes you to the next image & clicking last image takes you to the first image', () => {
    act(() => {
      render(<PhotoGallery images={[{'url': 'abc.com', 'caption': 'Picture 1'}, {'url': 'hey.com', 'caption': 'Picture 2'}]} />, container);
    });

    const image = container.querySelector('img');

    expect(container).toMatchSnapshot();
    expect(container.textContent).toBe('Picture 1');

    act( () => {
    image.dispatchEvent(new MouseEvent("click", { bubbles: true }));
    });

    expect(container).toMatchSnapshot();
    expect(container.textContent).toBe('Picture 2');   
    
    act( () => {
    image.dispatchEvent(new MouseEvent("click", { bubbles: true }));
    });

    expect(container.textContent).toBe('Picture 1');
  })
})