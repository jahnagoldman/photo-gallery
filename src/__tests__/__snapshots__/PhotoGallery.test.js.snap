// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhotoGallery clicking the image takes you to the next image & clicking last image takes you to the first image 1`] = `
<div>
  <div>
    <div
      class="PhotoContainer"
    >
      <img
        alt="new"
        src="abc.com"
      />
      <div
        class="PhotoCaption"
      >
        Picture 1
      </div>
    </div>
  </div>
</div>
`;

exports[`PhotoGallery clicking the image takes you to the next image & clicking last image takes you to the first image 2`] = `
<div>
  <div>
    <div
      class="PhotoContainer"
    >
      <img
        alt="new"
        src="hey.com"
      />
      <div
        class="PhotoCaption"
      >
        Picture 2
      </div>
    </div>
  </div>
</div>
`;

exports[`PhotoGallery renders with empty list of images 1`] = `
<div>
  <div>
    <div
      class="NoPhotos"
    >
      No photos!
    </div>
  </div>
</div>
`;

exports[`PhotoGallery renders with images 1`] = `
<div>
  <div>
    <div
      class="PhotoContainer"
    >
      <img
        alt="new"
        src="abc.com"
      />
      <div
        class="PhotoCaption"
      >
        hello
      </div>
    </div>
  </div>
</div>
`;

exports[`PhotoGallery renders without images 1`] = `
<div>
  <div>
    <div
      class="NoPhotos"
    >
      No photos!
    </div>
  </div>
</div>
`;
